pkg/server/server.go

```diff
@@ -244,10 +244,53 @@ func (s *Server) Run(ctx context.Context) error {
 					VirtualWorkspaceURL: s.Options.Extra.ShardVirtualWorkspaceURL,
 				},
 			}
-			if _, err := s.RootShardKcpClusterClient.Cluster(tenancyv1alpha1.RootCluster).TenancyV1alpha1().ClusterWorkspaceShards().Create(goContext(ctx), shard, metav1.CreateOptions{}); err != nil {
-				return err
+
+			if err := wait.PollInfiniteWithContext(goContext(ctx), time.Second, func(ctx context.Context) (bool, error) {
```

Comment:
Should we use an exponential backoff here?

Reply:
I don't think so. We do it on a server startup, so we want it to be quick. Backing off might delay server initialization.